interfaces/builtin/network_manager.go

```diff
@@ -0,0 +1,429 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"bytes"
+
+	"github.com/ubuntu-core/snappy/interfaces"
+)
+
+var networkManagerPermanentSlotAppArmor = []byte(`
+# Description: Allow operating as the NetworkManager service. Reserved because this
+#  gives privileged access to the system.
+# Usage: reserved
+
+capability net_admin,
+capability net_bind_service,
+capability net_raw,
+
+network netlink,
+network netlink raw,
+network netlink dgram,
+network bridge,
+network inet,
+network inet6,
+network packet,
+
+@{PROC}/@{pid}/net/ r,
+@{PROC}/@{pid}/net/** r,
+
+# used by sysctl, et al
+@{PROC}/sys/ r,
+@{PROC}/sys/net/ r,
+@{PROC}/sys/net/core/ r,
+@{PROC}/sys/net/core/** rw,
+@{PROC}/sys/net/ipv{4,6}/ r,
+@{PROC}/sys/net/ipv{4,6}/** rw,
+@{PROC}/sys/net/netfilter/ r,
+@{PROC}/sys/net/netfilter/** rw,
+@{PROC}/sys/net/nf_conntrack_max rw,
+
+# Needed for systemd's dhcp implementation
+@{PROC}/sys/kernel/random/boot_id r,
+
+/sys/devices/**/**/net/**/phys_port_id r,
+/sys/devices/**/**/net/**/dev_id r,
+/sys/devices/virtual/net/**/phys_port_id r,
+/sys/devices/virtual/net/**/dev_id r,
+
+/dev/rfkill rw,
+
+# Needed by the ifupdown plugin to check which interfaces can
+# be managed an which not.
+/etc/network/interfaces r,
+# Needed for systemd's dhcp implementation
+/etc/machine-id r,
+
+#include <abstractions/nameservice>
+
+# DBus accesses
+#include <abstractions/dbus-strict>
+
+dbus (send)
+   bus=system
+   path=/org/freedesktop/DBus
+   interface=org.freedesktop.DBus
+   member={Request,Release}Name
+   peer=(name=org.freedesktop.DBus),
+
+dbus (receive, send)
+   bus=system
+   path=/org/freedesktop/DBus
+   interface=org.freedesktop.DBus
+   member=GetConnectionUnixProcessID
+   peer=(label=unconfined),
+
+dbus (receive, send)
+   bus=system
+   path=/org/freedesktop/DBus
+   interface=org.freedesktop.DBus
+   member=GetConnectionUnixUser
+   peer=(label=unconfined),
+
+# Allow binding the service to the requested connection name
+dbus (bind)
+    bus=system
+    name="org.freedesktop.NetworkManager",
+
+# Allow traffic to/from our path and interface with any method
+dbus (receive, send)
+    bus=system
+    path=/org/freedesktop/NetworkManager{,/**}
+    interface=org.freedesktop.NetworkManager*,
+
+# Allow traffic to/from org.freedesktop.DBus for NetworkManager service
+dbus (receive, send)
+    bus=system
+    path=/org/freedesktop/NetworkManager{,/**}
+    interface=org.freedesktop.DBus.**,
+
+# Allow access to hostname system service
+dbus (send)
+    bus=system
+    path=/org/freedesktop/hostname1
+    interface=org.freedesktop.DBus.Properties
+    peer=(label=unconfined),
```

Comment:
We can use peer label rules here for the two rules zyga mentioned (and we should). If this is only for the service to talk to itself, you can use `label=@{profile_name}` (`@{profile_name}` is a special apparmor variable for the name of the profile of the current process) and if this is for connected client, perhaps this rule should be in connected slots instead of permanent slots.

Note, with 15.04 frameworks we said that dbus policy would be wide open and server policy would be wide-open and the framework-policy client policy would be the primary mediation point. However, with interfaces we have much more information to create precise rules for both server side and client side.